app/com/lynxanalytics/biggraph/graph_operations/Aggregate.scala

```diff
@@ -271,6 +271,29 @@ trait CompoundDoubleAggregator[From]
   def outputTypeTag(inputTypeTag: TypeTag[From]) = typeTag[Double]
 }
 
+// ElementwiseAggregator executes a given Double aggregator on Vector[Double] attributes,
+// padding them to equal lengths with the configured zero element if needed.
+class Elementwise[Intermediate](val agg: ItemAggregator[Double, Intermediate, Double])
+  extends ItemAggregator[Vector[Double], Vector[Intermediate], Vector[Double]] with Serializable {
+  def zeroElement: Double = 0
+  def zeroIntermediate: Intermediate = agg.zero
+  def zero = Vector[Intermediate]()
+  def merge(a: Vector[Intermediate], b: Vector[Double]) =
+    a.padTo(b.size, zeroIntermediate).zip(b.padTo(a.size, zeroElement)).map {
```

Comment:
Why do padding? I don't really see a good usecase where this is required, I mean when I naturally end up with different size vectors and I want to pad them with 0s. But I do see that this behavior can hide errors, i.e. I think the vectors should be equal size but they are not due to some upstream problem. I think in this case it's much better to get an error. If there is a really good usecase for this then I'd rather add a separate padding operation.

Reply:
> Why do padding? I don't really see a good usecase where this is required, I mean when I naturally end up with different size vectors and I want to pad them with 0s. But I do see that this behavior can hide errors, i.e. I think the vectors should be equal size but they are not due to some upstream problem. I think in this case it's much better to get an error. If there is a really good usecase for this then I'd rather add a separate padding operation.

I'm sorry, it's because of the zero value. At that point I don't know how wide the vectors are, so I can't create a proper zero vector. I create an empty vector instead and later pad it.

I guess an alternative would be to use `null` as the zero vector and special case it here and there. Would you prefer that?

Reply:
Yeah, I think that would be nicer. If you define sg like:
```
def fillIfNull(vector, size): Vector[Intermediate] {
  if (vector == null) ... create zero vector ...
  else ... assert size match and return vector ...
}
```
Then the code will even be simpler.

The only question is finalize. :) I guess you can just assert not null there.

Reply:
> Yeah, I think that would be nicer. If you define sg like:
> 
> ```
> def fillIfNull(vector, size): Vector[Intermediate] {
>   if (vector == null) ... create zero vector ...
>   else ... assert size match and return vector ...
> }
> ```

I've included the whole other vector as a parameter for better assertion messages.

> Then the code will even be simpler.

I may have made a mistake because it's not simpler. 😓 

> The only question is finalize. :) I guess you can just assert not null there.

In practice `finalize` should never see a `null`. If the set over which it's aggregating is empty, the result is always undefined. (Which is a bit weird. E.g. we may expect `count` to be 0.) So I didn't add an assertion. There will be an exception anyway if this comes up.